### PR TITLE
Gigasecond rewrite -- draft

### PR DESCRIPTION
**Pull request by kytrinyx**
I tried to make the instructions open to allow us to add kiloseconds and megaseconds (or even multiples of these, perhaps to give rough equivalents of things that we already have a sense for. An hour is roughly 4 kiloseconds. A day is roughly 100 kiloseconds. Etc.

### CODE REVIEW

File: exercises/gigasecond/introduction.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+And if we ever colonize Mars or some other planet, it's going to get even messier.
+If someone says "year" do they mean Earth-year or Mars-year?
+
+Instead, we could standardize on the metric system, and count everything in seconds.
+
+A recipe might explain that you need to let the brownies cook in the oven for two kiloseconds (that's two thousand seconds).
+
+Perhaps you and your family would travel to somewhere exotic for two megaseconds (that's two million seconds).
+
+And if you and your spouse were married for _a thousand million_ seconds, you would celebrate your one gigasecond anniversary.
```
Comment (ErikSchierboom) [inline exercises/gigasecond/introduction.md:22]: Great!

**Review by ErikSchierboom — APPROVED** [review]

File: exercises/gigasecond/introduction.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Introduction
+
+The way we count minutes and hours comes from ancient Babylon, where they used 60 as the basis for their number system.
+So we have 60 seconds in a minute, and 60 minutes in an hour.
+But then it starts getting complicated.
+We have 24 hours in a day, 7 days in a week, and months?
+Well, for months it depends not only on which month it is, but also what calendar is used in the country you live in.
+
+At least there are always 365 days in a year.
+Well, except if it's a leap year and have 366 days.
```
Comment (iHiD) [inline exercises/gigasecond/introduction.md:10]:
```suggestion
```

File: exercises/gigasecond/introduction.md (hunk 1 above)
Comment (iHiD) [inline exercises/gigasecond/introduction.md:22]:
```suggestion
Instead, we could standardize on the metric system, and count everything in seconds:

- A food recipe might explain that you need to let the brownies cook in the oven for two kiloseconds (that's two thousand seconds).
- Perhaps you and your family would travel to somewhere exotic for two megaseconds (that's two million seconds).
- And if you and your spouse were married for _a thousand million_ seconds, you would celebrate your one gigasecond anniversary.
```

**Comment by iHiD** [comment]
General feedback:
- I think we should aim to have the introduction slightly shorter and more compressed than this draft. I'm thinking that adding **too** much detail might make it harder for non-native speakers. So I think somewhere in between the 1-line and this ~10 line version? My suggested changes to `introduction.md` make this about what feels right to me - maybe still a little on the long side, but more within what feels ok. 
- I also think the length of the introduction should probably match the complexity of the exercise. we don't want people spending longer reading the story than solving the exercise (which is maybe possible here).
- I really like that the story isn't a made up story, but some interesting idea with some examples. I think that's a good way of approaching lots of these.

---

Specific feedback:
> "Your task is to determine the date and time that a metric timespan has passed."

I really struggle to parse this!

"that a metric timespan has passed" - in my head this is like a timespan passing something. I think it means this:

"Your task is to determine the date and time one gigasecond after a certain date."

If it does, could we use that instead. If it doesn't then I don't understand the task :)

**Comment by kytrinyx** [comment]
I tweaked it a bit more. Would you give it another look?

**Comment by kytrinyx** [comment]
> If it does, could we use that instead. If it doesn't then I don't understand the task :)

It works.

I was trying to figure out how to make the instructions flexible so we could add kiloseconds and megaseconds into the mix without messing up the instructions for tracks that only have one gigasecond. But I have come to the conclusion that that is a problem for Future Katrina!

**Comment by iHiD** [comment]
>I was trying to figure out how to make the instructions flexible so we could add kiloseconds and megaseconds into the mix without messing up the instructions for tracks that only have one gigasecond. 

Ah - makes sense!

**Review by iHiD — APPROVED** [review]
Yep - I like this!

**Comment by kytrinyx** [comment]
This has been opened on Jeremy's fork.

**Pull request closed without merging**